kernel/src/main/java/org/kframework/kompile/DefinitionParsing.java

```diff
@@ -285,6 +285,17 @@ protected Definition resolveConfigBubbles(Definition definition, Module defaultC
         errors = java.util.Collections.synchronizedSet(Sets.newHashSet());
         caches = loadCaches();
 
+        // Check to ensure that the k cell is not duplicated.
+        // Note: this is done before the configuration bubbles are resolved because it allows us to use location
+        // information from the sentences that declare the configuration cells.
+        List<Bubble> KCellDeclarations = stream(definition.getAllConfigDecls())
+                .filter(s -> s.contents().contains("<k>") && s.contents().contains("</k>"))
+                .collect(Collectors.toList());
+        if (KCellDeclarations.size() > 1) {
+            KCellDeclarations.stream().forEach(k ->
+                errors.add(KEMException.compilerError("Multiple K cell declarations detected. Only one declaration is allowed.", k)));
+        }
+
```

Comment:
```
configuration <k> "<k>" </k>
```
If you do this at the text level, you will fail for such an example.

But you got the right idea. Below this function you have 
```java
Definition defWithParsedConfigs = ...

// replace config bubbles with the generated syntax and rules
```
You have to write a [K visitor](https://github.com/runtimeverification/k/blob/master/kernel/src/main/java/org/kframework/compile/CloseCells.java#L102) over `defWithParsedConfigs` that looks for `#token("k","#CellName")`

Here is an example configuration
```
configuration <k> $PGM:KItem </k>
              <state> .Map </state>
Configuration(#cells(#configCell(#token("k","#CellName"),#cellPropertyListTerminator(.KList),#SemanticCastToKItem(#token("$PGM","KConfigVar")),#token("k","#CellName")),#configCell(#token("state","#CellName"),#cellPropertyListTerminator(.KList),`.Map`(.KList),#token("state","#CellName"))),#token("true","Bool"),[org.kframework.attributes.Location(Location(11,19,12,40)), org.kframework.attributes.Source(Source(/home/radu/work/test/test.k)), org.kframework.definition.Production(syntax #RuleContent ::= #RuleBody [klabel(#ruleNoConditions), symbol])])
```

Reply:
Ah, Thanks!